src/mdns/mdns.hpp

```diff
@@ -148,6 +151,9 @@ class Publisher : private NonCopyable
         AddressList mAddresses;      ///< IP6 addresses.
         uint32_t    mNetifIndex = 0; ///< Network interface.
         uint32_t    mTtl        = 0; ///< Host TTL.
+
+        void AddAddress(Ip6Address aAddress) { Publisher::AddAddress(mAddresses, std::move(aAddress)); }
+        void RemoveAddress(const Ip6Address &aAddress) { Publisher::RemoveAddress(mAddresses, aAddress); }
```

Comment:
I noticed `AddAddress()` expects a `Ip6Address aAddress` input but `RemoveAddress()` a `const` reference as `const Ip6Address &aAddress`.

Any reason for this? Can they both get `const Ip6Address &aAddress`?

Reply:
`AddAddress` takes the parameter as value so it can leverage the move constructor of `Ip6Address` to save copying. Although as of today, `Ip6Address`'s move constructor does the same as its copy constructor, it'll save some copying if some day `Ip6Address` stores some data on the heap.

Reply:
Thanks.  Since `Ip6Address` today does not have a move constructor, it is always copied (it seems to follow value semantics as a `union` of 16 bytes). 

So if I understand correctly, with the use of `Ip6Address aAddress` as input, we may end up doing 3 copies of `Ip6Address`:
- When `AddAddress()` is first called, we can have one copy for `Ip6Address aAddress`. 
- Then when we call the `static` `Publisher::AddAddress()` we can have another copy.
- And finally when we `push_back` into the list.
(compiler may optimize some of this, but it does not have to).

I think if we use `const Ip6Address &aAddress`, we end up doing one copy (when we `push_back`)?
If we add move semantics later for `Ip6Address` (which I think probably unlikely?), we can optimize the code then by providing a new flavor of `AddAddress(Ip6Address &&aAddress)` working with `&&` reference?

Reply:
Thanks for pointing it out! I think you're right that `std::move` doesn't fit well here. I've updated the code to pass in const reference.